internal/characters/bennett/plunge.go

```diff
@@ -0,0 +1,164 @@
+package bennett
+
+import (
+	"fmt"
+
+	"github.com/genshinsim/gcsim/internal/frames"
+	"github.com/genshinsim/gcsim/pkg/core/action"
+	"github.com/genshinsim/gcsim/pkg/core/attacks"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/core/player"
+)
+
+var highPlungeFrames []int
+var lowPlungeFrames []int
+
+const collisionHitmark = 30
+const highPlungeHitmark = 35
+const lowPlungeHitmark = 33
+
+func init() {
+	// high_plunge -> x
+	highPlungeFrames = frames.InitAbilSlice(68)
+	highPlungeFrames[action.ActionAttack] = 47
+	highPlungeFrames[action.ActionSkill] = 46
+	highPlungeFrames[action.ActionBurst] = 47
+	highPlungeFrames[action.ActionDash] = 38
+	highPlungeFrames[action.ActionWalk] = 67
+	highPlungeFrames[action.ActionSwap] = 54
+
+	// low_plunge -> x
+	lowPlungeFrames = frames.InitAbilSlice(67)
+	lowPlungeFrames[action.ActionAttack] = 45
+	lowPlungeFrames[action.ActionSkill] = 46
+	lowPlungeFrames[action.ActionBurst] = 46
+	lowPlungeFrames[action.ActionDash] = 36
+	lowPlungeFrames[action.ActionWalk] = 66
+	lowPlungeFrames[action.ActionSwap] = 53
+}
+
+// High Plunge attack damage queue generator
+// Use the "collision" optional argument if you want to do a falling hit on the way down
+// Default = 0
+func (c *char) HighPlungeAttack(p map[string]int) (action.Info, error) {
+	switch c.Core.Player.Airborne() {
+	case player.AirborneVenti:
+		return action.Info{}, fmt.Errorf("bennett plunge while airborne due to venti is unimplemented due to lack of frame data. Please see https://docs.gcsim.app/mechanics/frames for how to contribute")
+	case player.AirborneXianyun:
+		return c.highPlungeXY(p)
+	default:
+		return action.Info{}, fmt.Errorf("bennett high_plunge can only be used while airborne")
+	}
+}
+
+func (c *char) highPlungeXY(p map[string]int) (action.Info, error) {
+	collision, ok := p["collision"]
+	if !ok {
+		collision = 0 // Whether or not Xiao does a collision hit
+	}
+
+	if collision > 0 {
+		c.plungeCollision(collisionHitmark)
+	}
+
+	poiseDMG := 150.0
```

Comment:
nit: i assume compiler optimizes this variable away. doesn't look like it gets changed anywhere. should probably get rid of the variable here